pkg/nsx/services/nsxserviceaccount/store.go

```diff
@@ -0,0 +1,130 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package nsxserviceaccount
+
+import (
+	"errors"
+	"strconv"
+	"sync"
+
+	mpmodel "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+	nsxutil "github.com/vmware-tanzu/nsx-operator/pkg/nsx/util"
+)
+
+func queryPrincipalIdentity(service *NSXServiceAccountService, wg *sync.WaitGroup, fatalErrors chan error) {
```

Comment:
You can rebase the latest vpc_dev branch, and reuse function InitializeResourceStore, refer to [https://github.com/vmware-tanzu/nsx-operator/blob/vpc_dev/pkg/nsx/services/securitypolicy/firewall.go#L56](url)

Reply:
This is MP resource and has some gap to use common store. I'll refactor queryClusterControlPlane.

Reply:
Refactored.